packages/nimbus/src/components/date-picker/date-picker.stories.tsx

```diff
@@ -0,0 +1,2509 @@
+      const dateSegments = canvas.getAllByRole("spinbutton");
+
+      // First segment should be focusable with Tab
+      await userEvent.tab();
+      await expect(dateSegments[0]).toHaveFocus();
+
+      // Should be able to navigate between segments with arrow keys
+      await userEvent.keyboard("{ArrowRight}");
+      if (dateSegments.length > 1) {
+        await expect(dateSegments[1]).toHaveFocus();
+      }
+
+      // Should be able to navigate to calendar button eventually
+      await userEvent.tab();
+      await userEvent.tab(); // May need multiple tabs depending on segments
+      const calendarButton = canvas.getByRole("button", { name: /calendar/i });
+      // Check if calendar button has focus or if we need more tabs
+      if (!calendarButton.matches(":focus")) {
+        await userEvent.tab();
+      }
+      await expect(calendarButton).toHaveFocus();
+    });
+
+    await step("Date segments accept keyboard input", async () => {
+      const dateSegments = canvas.getAllByRole("spinbutton");
+      const firstSegment = dateSegments[0];
+
+      // Focus first segment and type a value
+      await userEvent.click(firstSegment);
+      await userEvent.keyboard("12");
+
+      // Segment should have the typed value
+      await expect(firstSegment).toHaveAttribute("aria-valuenow", "12");
+
+      // Clear segments for next tests by selecting all and deleting
+      for (const segment of dateSegments) {
+        await userEvent.click(segment);
+        await userEvent.keyboard("{Control>}a{/Control}");
+        await userEvent.keyboard("{Delete}");
+      }
+    });
+
+    await step("Calendar popover opens and closes correctly", async () => {
+      const calendarButton = canvas.getByRole("button", { name: /calendar/i });
+
+      // Initially, calendar should not be visible (check in entire document since popover is portaled)
+      let calendar = within(document.body).queryByRole("application");
+      await expect(calendar).not.toBeInTheDocument();
+
+      // Click calendar button to open popover
+      await userEvent.click(calendarButton);
+
+      // Wait for calendar to appear (it's rendered in a portal)
+      await waitFor(async () => {
+        calendar = within(document.body).queryByRole("application");
+        await expect(calendar).toBeInTheDocument();
+      });
+
+      // Should be able to close with Escape key
+      await userEvent.keyboard("{Escape}");
+
+      // Wait for calendar to disappear
+      await waitFor(async () => {
+        calendar = within(document.body).queryByRole("application");
+        await expect(calendar).not.toBeInTheDocument();
+      });
+    });
+
+    await step("Clear button functionality works correctly", async () => {
+      const dateSegments = canvas.getAllByRole("spinbutton");
+
+      // Initially, clear button should be disabled (no value selected)
+      const clearButton = canvas.getByRole("button", { name: /clear/i });
```

Comment:
I kind of like the pattern of persisting the cliear button and disabling it if there's no input value.  Is it something we should consider for `Select` and `ComboBox`?